source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -333,23 +333,33 @@ export class CardDesignerSurface {
         if (this.isPreviewMode) {
             cardToRender.onExecuteAction = (action: Adaptive.Action) => {
                 const actionType = action.getJsonTypeName();
-                let message = `Action executed: "${actionType}" title: "${action.title}"`;
+                const message = `Action executed: "${actionType}" title: "${action.title}"`;
+                let verb : string;
+                let url : string;
+                let data : string;
+
+                switch (actionType)
+                {
+                    case Adaptive.OpenUrlAction.JsonTypeName:
+                        const actionOpenUrl = <Adaptive.OpenUrlAction>action;
```

Comment:
with the optional chaining operator (?.), it will have less code
```
const url = (<Adaptive.OpenUrlAction>action)?.url;
const verb = (<Adaptive.ExecuteAction>action)?.verb;
```